avm/ptn/lz/sub-vending/main.json

```diff
@@ -508,7 +531,7 @@
                 "workload": "[parameters('subscriptionWorkload')]",
                 "displayName": "[parameters('subscriptionDisplayName')]",
                 "billingScope": "[parameters('subscriptionBillingScope')]",
-                "additionalProperties": "[if(and(not(empty(parameters('subscriptionTenantId'))), not(empty(parameters('subscriptionOwnerId')))), createObject('subscriptionTenantId', parameters('subscriptionTenantId'), 'subscriptionOwnerId', parameters('subscriptionOwnerId')), createObject())]"
+                "additionalProperties": "[variables('formattedAdditionalProperties')]"
```

Comment:
From what I understand of this RP, moving this to here makes it a WORM style operation and therefore you cannot update the Management Group Association after the initial creation.

Hence why we do it as a separate resource/operation today to ensure you can move a subscription later down the line, if you so wish.